core/app/[locale]/(default)/(auth)/login/_actions/login.ts

```diff
@@ -42,11 +43,17 @@ export const login = async (_lastResult: SubmissionResult | null, formData: Form
       });
     }
 
-    if (error instanceof Error) {
-      return submission.reply({ formErrors: [error.message] });
+    if (error instanceof AuthError) {
+      if (error.name === 'CallbackRouteError') {
+        if (error.cause && error.cause.err?.message.includes('Invalid credentials')) {
+          return submission.reply({ formErrors: [t('Form.invalidCredentials')] });
+        }
+      }
+
+      return submission.reply({ formErrors: [error.name] });
```

Comment:
Can we remove the nested conditionals?
```suggestion
    if (
      error instanceof AuthError &&
      error.name === 'CallbackRouteError' &&
      error.cause &&
      error.cause.err?.message.includes('Invalid credentials')
      ) {
          return submission.reply({ formErrors: [t('Form.invalidCredentials')] });
      }

      return submission.reply({ formErrors: [error.name] });
```

Reply:
Fixed!